Controller/Adminhtml/Menu/Save.php

```diff
@@ -194,9 +194,9 @@ public function execute()
 
                     $nodeObject->setMenuId($menu->getMenuId());
                     $nodeObject->setTitle($node['title']);
-                    $nodeObject->setIsActive((int) ($node['is_active'] ?? 0));
-                    $nodeObject->setLevel($level);
-                    $nodeObject->setPosition($position);
+                    $nodeObject->setIsActive($node['is_active'] ?? '0');
```

Comment:
I can see that `is_active` methods are declared to receive `int` type as parameter and returns `int` as well, not sure why was this changed to be `string`? (applies to level and position)

Reply:
I made this change because when Magento loads menu and node objects, the `integer` data inside them get loaded as `string` (even if they were initially stored as integers).  
And when Magento tries to set `integer` data inside those object attributes, it will label the object as an [object with data changes](https://github.com/magento/magento2/blob/2.4.0/lib/internal/Magento/Framework/Model/AbstractModel.php#L356-L358).  
That's because `$this->_data[$key] !== $value` will be `true`, due to data type differences, even if the values are the same.

Thus, when Magento tries to save the object, it will treat it as [modified](https://github.com/magento/magento2/blob/2.4.0/lib/internal/Magento/Framework/Model/ResourceModel/Db/AbstractDb.php#L405-L410), and will it [update](https://github.com/magento/magento2/blob/2.4.0/lib/internal/Magento/Framework/Model/ResourceModel/Db/AbstractDb.php#L418-L420) it even without actual data changes.
That also means that the page cache will get unnecessarily invalidated.

I think this is currently the simplest workaround available to the issue, since that the attributes are not type hinted.

Reply:
You're right, I have tested a few cases and indeed this workaround resolves the issue